xdsl/irdl.py

```diff
@@ -325,6 +325,36 @@ def irdl_to_attr_constraint(
 
 class IRDLOperation(Operation):
 
+    @staticmethod
+    def get_def() -> OpDef:
+        ...
+
+    def __init__(
+        self: IRDLOperation,
+        operands: Sequence[SSAValue | Operation
+                           | Sequence[SSAValue | Operation] | None]
+        | None = None,
+        result_types: Sequence[Attribute | Sequence[Attribute]]
+        | None = None,
+        attributes: Mapping[str, Attribute | None] | None = None,
+        successors: Sequence[Block] | None = None,
+        regions: Sequence[Region | Sequence[Operation] | Sequence[Block]
+                          | Sequence[Region | Sequence[Operation]
+                                     | Sequence[Block]]]
+        | None = None):
+        if operands is None:
+            operands = []
+        if result_types is None:
+            result_types = []
+        if attributes is None:
+            attributes = {}
+        if successors is None:
+            successors = []
+        if regions is None:
+            regions = []
+        irdl_op_init(self, self.get_def(), operands, result_types, attributes,
+                     successors, regions)
+
     @classmethod
     def build(
```

Comment:
This is just left here because you did not want to refactor all the dialects yet, right? Is there another reason to still have build and create?

Reply:
So renaming them to one name requires some discussion for me. Because they look similar, but have slight differences in the semantics of their arguments, and are both intended for usages where you don't know the type when writing the code. Happy to discuss it, but sounds like another PR's material to me indeed!

Basically, if I rename `IRDLOperation.build` to `IRDLOperation.create`, the behaviour of `cls.create([...])` now depends on `cls` being an IRDLOperation subclass or a plain Operation (or yet another thing?).

Reply:
I was thinking of removing both and replacing them with the two different flavors of init (one on Operation, the other on IRDLOperation). But maybe that is just me being out of the loop an not knowing all the issues/considerations that are relevant here.

Reply:
Why doesn't this work with `__new__`? We could override `__new__` to do what create currently does, right? (i.e. build and return the object)

Reply:
We would have the same problem.
If we do override `__new__` or `__init__`, we end up not being able to create an operation given a list of operands, results, ....
So that's why we use manually `__new__` and `Operation.__init__` in the `create` function.